chain/solana/provider/rpcclient/client.go

```diff
@@ -0,0 +1,310 @@
+package rpcclient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/avast/retry-go/v4"
+	sollib "github.com/gagliardetto/solana-go"
+	solrpc "github.com/gagliardetto/solana-go/rpc"
+	"github.com/gagliardetto/solana-go/rpc/jsonrpc"
+)
+
+// sendConfig defines the configuration for sending transactions.
+type sendConfig struct {
+	// RetryAttempts specifies how many times to retry the send operations. This is used in all
+	// underlying calls to the RPC client.
+	// Setting this value to 0 will cause the operation to retry indefinitely until it succeeds.
+	RetryAttempts uint
+	// RetryDelay specifies the delay between retries.
+	RetryDelay time.Duration
+	// FixedRetryAttempts provides a fixed number of attempts for certain retryable operations.
+	// Currently this is only used for confirming transactions where we want to retry a fixed
+	// number of times independent of the specified retry attempts.
+	FixedRetryAttempts uint
```

Comment:
From reading the api, it is not obvious the difference between `RetryAttempts` and `FixedRetryAttempts` , even the comments are slightly confusing, both `RetryAttempts` and `FixedRetryAttempts` are setting the number of retries to a "fixed" value. In reality they are being used in different calls.

Should we rename them appropriately?
```suggestion`
	ConfirmTransactionRetryAttempts uint
```

Reply:
I'm wary of prescribing 'use' to these variable names because it could change in the future. 

However since we don't expose this field out as configurable, it would be ok to named it `ConfirmXXX` because we change it out without impacting consumers.

I'll make a change as see how it looks